agentai/api.py

```diff
@@ -1,31 +1,36 @@
 """
 API functions for the agentai package
 """
-from typing import Callable
+from typing import Callable, List
 
 import openai
 import requests
 from loguru import logger
 from tenacity import retry, retry_if_exception_type, stop_after_attempt
 
-from .conversation import Conversation
+from .conversation import Conversation, Message
 from .openai_function import ToolRegistry
 
 logger.disable(__name__)
 
+class InvalidOutput(Exception):
+    pass
 
-@retry(retry=retry_if_exception_type(ValueError), stop=stop_after_attempt(3))
+retry_on_exceptions = retry_if_exception_type((InvalidOutput, requests.HTTPError, KeyError))
```

Comment:
SQLErrors?

Reply:
Hmm. Then maybe we need to find a way to retry on anything other than ValueError.

Reply:
Done.